src/call.cpp

```diff
@@ -985,4 +985,46 @@ void DarlingServer::Call::S2CPerform::processCall() {
 	_sendReply(code);
 };
 
+void DarlingServer::Call::SetExecutablePath::processCall() {
+	int code = 0;
+	int fullLength = 0;
+
+	if (auto thread = _thread.lock()) {
+		if (auto process = thread->process()) {
+			std::string tmpstr;
+			tmpstr.resize(_body.buffer_size);
+			if (!process->readMemory((uintptr_t)_body.buffer, tmpstr.data(), _body.buffer_size, &code)) {
+				code = -code;
+			}
+			process->setExecutablePath(tmpstr.c_str());
+		} else {
+			code = -ESRCH;
+		}
+	} else {
+		code = -ESRCH;
+	}
+
+	_sendReply(code == 0 ? fullLength : code);
+}
+
+void DarlingServer::Call::GetExecutablePath::processCall() {
+	int code = 0;
+	uint64_t fullLength;
+
+	if (auto maybeTargetProcess = processRegistry().lookupEntryByNSID(_body.pid)) {
+		auto targetProcess = *maybeTargetProcess;
+		auto path = targetProcess->executablePath();
+		auto len = std::max(path.length() + 1, _body.buffer_size);
+		if (!targetProcess->writeMemory((uintptr_t)_body.buffer, path.data(), len, &code)) {
+			code = -code;
+		}
+		fullLength = path.length();
+		
+	} else {
+		code = -ESRCH;
+	}
+
+	_sendReply(code == 0 ? fullLength : code);
```

Comment:
Like I said earlier, the reply code is only meant to return error codes or `0` for success. If you decide to add a return value to the call (like in my suggestion for `generate-rpc-wrappers.py`):
```suggestion
	_sendReply(code, fullLength);
```